src/proxy.rs

```diff
@@ -100,20 +104,20 @@ impl Proxy {
     /// # fn run() -> Result<(), Box<::std::error::Error>> {
     /// let target = reqwest::Url::parse("https://my.prox")?;
     /// let client = reqwest::Client::builder()
-    ///     .proxy(reqwest::Proxy::custom(move |url| {
-    ///         if url.host_str() == Some("hyper.rs") {
-    ///             Some(target.clone())
-    ///         } else {
-    ///             None
-    ///         }
-    ///     }))
+    ///     .proxy(reqwest::Proxy::custom(|url| url.host() == Some("hyper.rs"),
+    ///                                   "http://proxy.custom")?)
     ///     .build()?;
     /// # Ok(())
     /// # }
     /// # fn main() {}
-    pub fn custom<F>(fun: F) -> Proxy
-    where F: Fn(&Url) -> Option<Url> + Send + Sync + 'static {
-        Proxy::new(Intercept::Custom(Custom(Arc::new(fun))))
+    pub fn custom<F, U: IntoUrl>(fun: F, url: U) -> ::Result<Proxy>
+    where F: Fn(&Uri) -> bool + 'static + Send + Sync {
```

Comment:
I didn't realize this change before. So, this changes it such that the function only returns true or false, and if true, then use the proxy at the provided URL. The behavior it is replacing is that the function would return both a) if to intercept at all, and b) the URL to use as a proxy. Is the new behavior easier to understand?

For those that had custom functions that would return different URLs conditionally, I suppose that can be done still, just be configuring multiple `Proxy::custom`s?

(Would also be a breaking change, so if it is better, this would need to be part of 0.9).

Reply:
Yes indeed.

I can change it back to previous behavior if really needed (I don't think I'll change the logic in hyper-proxy) but I (personally) find it easier to understand this way (*proxy = intercept + uri*).

I should definitely have underlined all these changes though, sorry about that!